packages/nextjs/src/editing/utils.ts

```diff
@@ -0,0 +1,302 @@
+import {
+  DesignLibraryRenderPreviewData,
+  EDITING_ALLOWED_ORIGINS,
+  EditingRenderQueryParams,
+  isDesignLibraryMode,
+  PREVIEW_KEY,
+  QUERY_PARAM_EDITING_SECRET,
+} from '@sitecore-content-sdk/core/editing';
+import { DEFAULT_VARIANT } from '@sitecore-content-sdk/core/personalize';
+import { SITE_KEY } from '@sitecore-content-sdk/core/site';
+import { NextApiRequest } from 'next';
+import { NextRequest } from 'next/server';
+import {
+  EDITING_PASS_THROUGH_HEADERS,
+  QUERY_PARAM_VERCEL_PROTECTION_BYPASS,
+  QUERY_PARAM_VERCEL_SET_BYPASS_COOKIE,
+} from './constants';
+import { IncomingHttpHeaders } from 'http';
+import { SERVER_PROPS_ID, STATIC_PROPS_ID } from 'next/constants';
+import { NativeDataFetcher } from '@sitecore-content-sdk/core';
+import { getAllowedOriginsFromEnv } from '@sitecore-content-sdk/core/utils';
+
+/**
+ * Gets editing secret value from request
+ * @param {NextApiRequest | NextRequest} req incoming request
+ * @returns {string | undefined} editing secret value if present
+ */
+export const getEditingSecretFromRequest = (req: NextApiRequest | NextRequest) => {
+  const reqQuery = (req as NextApiRequest).query;
+  const reqUrl = (req as NextRequest).url;
+
+  let secret = undefined;
+
+  if (reqQuery) {
+    // pages router
+    secret = reqQuery[QUERY_PARAM_EDITING_SECRET];
+  } else if (reqUrl) {
+    // app router
+    const url = new URL(reqUrl);
+    secret = url.searchParams.get(QUERY_PARAM_EDITING_SECRET);
+  }
+
+  return secret;
+};
+
+/**
+ * Parses query string and its parameters to required editing parameters
+ * @param {{ [key: string]: string | undefined }} query query string values
+ * @returns {EditingRenderQueryParams} editing parameters
+ */
+export const getEditingParams = (query: {
```

Comment:
we re-form query params into Editing params, maybe `mapEditingParams` name would be more clear